File: DSA/RAJAS-BAGGA/Day_15_Question_1.java
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode() {
    }

    TreeNode(int val) {
        this.val = val;
    }

    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

public class Day_15_Question_1 {
    static int count = 0;
    static int max = 0;
    public static int maxHeight(TreeNode root) {
        if (root == null)
            return 0;
        traverse(root);
        return max;
    }
    
    public static void traverse(TreeNode root) {
        if (root == null)
            return;
        
        ++count;
        traverse(root.left);
        traverse(root.right);
        max = Math.max(max, count);
        --count;
    }
    public static void main(String[] args) {
        TreeNode root = new TreeNode(3);
        TreeNode n1 = new TreeNode(2);
        TreeNode n2 = new TreeNode(5);
        TreeNode n3 = new TreeNode(1);
        TreeNode n4 = new TreeNode(4);
        TreeNode n5 = new TreeNode(6);
        
        root.left = n1;
        root.right = n2;
        n1.left = n3;
        n2.left = n4;
        n2.right = n5;
        System.out.println(maxHeight(root));
    }
}

File: DSA/Vikash_kumar/Day_15_Question_1.py
Day15-Question1-
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def height_of_binary_tree(root):
    if not root:
        return -1
    
    left_height = height_of_binary_tree(root.left)
    right_height = height_of_binary_tree(root.right)
    
    return 1 + max(left_height, right_height)

root = TreeNode(3)
root.left = TreeNode(2)
root.left.left = TreeNode(1)
root.right = TreeNode(5)
root.right.left = TreeNode(4)
root.right.right = TreeNode(6)

height = height_of_binary_tree(root)
print("Height of the binary tree:", height)  
